frontend/src/scenes/surveys/surveysLogic.tsx

```diff
@@ -48,22 +53,95 @@ export const surveysLogic = kea<surveysLogicType>([
         setSearchTerm: (searchTerm: string) => ({ searchTerm }),
         setSurveysFilters: (filters: Partial<SurveysFilters>, replace?: boolean) => ({ filters, replace }),
         setTab: (tab: SurveysTabs) => ({ tab }),
+        loadNextPage: true,
+        loadNextSearchPage: true,
     }),
     loaders(({ values }) => ({
-        surveys: {
-            __default: [] as Survey[],
+        data: {
+            __default: {
+                surveys: [] as Survey[],
+                surveysCount: 0,
+                searchSurveys: [] as Survey[],
+                searchSurveysCount: 0,
+            },
             loadSurveys: async () => {
-                const responseSurveys = await api.surveys.list()
-                return responseSurveys.results
+                const response = await api.surveys.list()
+                return {
+                    ...values.data,
+                    surveys: response.results,
+                    surveysCount: response.count,
+                }
```

Comment:
this block is repeating, we can just make a method and pass the `response` and `values.data` and return the final merged object

Reply:
they are all slightly different than each other. i.e. this one deals with `surveys` and `surveysCount`, and the next, `loadNextPage` merges those results.

however the other ones (for search) deal with other properties, `searchSurveys` and `searchSurveysCount`.

while we can create a method, because the properties are dynamic, i think it might be worse to follow the logic

Reply:
if a method takes params and just merges them, all that matters is that the method has a clear name and the params passed to it, so respectfully disagree with the 'it might be worse to follow the logic'
it's much easier to change things in one place, and keep it consistent across refactoring and debugging.

Reply:
I added separate methods for merning survey data and search survey data now, let me know what you think